test/jdk/javax/management/remote/mandatory/connection/DeadLockTest.java

```diff
@@ -46,21 +46,23 @@ public class DeadLockTest {
     public static void main(String[] args) {
         System.out.println(">>> test on a client notification deadlock.");
 
-        boolean ok = true;
+        boolean fail = false;
         for (int i = 0; i < protocols.length; i++) {
             try {
                 test(protocols[i]);
             } catch (Exception e) {
+                fail = true; // any one protocol failure, fails the test
```

Comment:
```suggestion
                fail = true; // any one protocol failure fails the test
```

Reply:
I actually think it's more readable with the comma.
If there is (one protocol failure), then that (fails the test).
Without the comma, "failure fails" runs together, but the failure did not fail, it was a perfectly good failure.  Pause for breath. What do we do now?  Well, experiencing that kind of problem, fails the test.

Extended discussions on language style, from the test that brought you "listner" and "should no block". 8-)

Reply:
The best way to get to the right answer here is simplify to the subject and verb: "failure fails". You don't put a comma between the subject and the verb, unless is something more much complex like "a failure, for which there can be more than one, fails the test". I think the reason you feel it reads better with the comma is because of the repetition of "fail". Would you still want a comma if the sentence was "any one protocol error fails the test"? I assume no, but the sentence is structurally identical.

Reply:
Right, it is the repetition that makes me want the comma.  There are other ways of phrasing this which would not need the comma.  Even then, I still might introduce one to imply a pause, which I still think helps make it unambiguous on first read, without making it "...causes the test to fail" which is unnecessarily lengthy.  It's also a comment buried in a test, not front page news.

Reply:
My 8th grade grammar teacher took no pity on students with "comma-itis" as he called it. They got penalized harshly for gratuitous use of commas. He definitely got through to me though.

Reply:
I think he should meet my parents, Jay-Z and Beyonce.